### ISSUE TITLE
[BUG] Player not found in cache

### ISSUE THREAD

**Issue by skaen** [K4ryuu/K4-System#94]
CSS API: 142
"K4-System" (3.0.0 (release)) by K4ryuu

```
2023-12-29 08:43:40.866 +03:00 [EROR] (cssharp:Core) Error invoking callback
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Collections.Generic.KeyNotFoundException: Player with ID 0 not found in cache
   at K4System.PlayerCache`1.get_Item(CCSPlayerController controller)
   at K4System.ModuleStat.SavePlayerStatCache(CCSPlayerController player, Boolean remove)
   at K4System.ModuleStat.<Initialize_Events>b__4_10(EventPlayerDisconnect event, GameEventInfo info)
   at InvokeStub_GameEventHandler`1.Invoke(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at CounterStrikeSharp.API.Core.FunctionReference.<>c__DisplayClass3_0.<.ctor>b__0(fxScriptContext* context) in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Core/FunctionReference.cs:line 82
2023-12-29 08:43:40.870 +03:00 [EROR] (cssharp:Core) Error invoking callback
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Collections.Generic.KeyNotFoundException: Player with ID 0 not found in cache

   at K4System.PlayerCache`1.get_Item(CCSPlayerController controller)
   at K4System.ModuleRank.SavePlayerRankCache(CCSPlayerController player, Boolean remove)
   at K4System.ModuleRank.<>c__DisplayClass5_0.<Initialize_Events>b__15(EventPlayerDisconnect event, GameEventInfo info)
   at InvokeStub_GameEventHandler`1.Invoke(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at CounterStrikeSharp.API.Core.FunctionReference.<>c__DisplayClass3_0.<.ctor>b__0(fxScriptContext* context) in /home/runner/work/CounterStrikeSharp/CounterStrikeSharp/managed/CounterStrikeSharp.API/Core/FunctionReference.cs:line 82

```
[log-all20231229.txt](https://github.com/K4ryuu/K4-System/files/13791386/log-all20231229.txt)

**Comment by K4ryuu** [comment]
Greetings,
At the plugin startup did you have any mismatches with the MySQL validator?

**Comment by K4ryuu** [comment]
I found and solved the problem. Thank you for the feedback!
It will be available in the upcoming release (3.0.1)
